### ISSUE TITLE
support ruby ERB templates

### ISSUE THREAD

**Issue by yeoldegrove** [koalaman/shellcheck#1721]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC1009,SC1073,SC1072
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7
- [x] The rule's wiki page does not already cover this
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
cat test.sh.erb
#!/bin/bash
<% if File.exist?('/bin/bash') %>
echo "Bash exists"
<% else %>
echo "Bash is missing"
<% end %>
```

#### Here's what shellcheck currently says:

In test.sh line 2:
<% if File.exist?('/bin/bash') %>
^-- SC1009: The mentioned parser error was in this simple command.
                                ^-- SC1073: Couldn't parse this redirection.
                                 ^-- SC1072:  Fix any mentioned problems and try again.

#### Here's what I wanted or expected to see:

Shellcheck should understand that this is a ruby ERB template and ignore everything in the brackets:

```
<% ... %>
```

**Comment by matthewpersico** [comment]
But consider how many other template systems are out there. Is `shellcheck` supposed to recognize them all, with their various set-off syntaxes? And, how would this work anyway? If bash DNE, how do lines 1 and 5 get executed? Perhaps a more real world example could help here? Probably easier here to not put real shell code in this, but extract said code to external script and just run that:
```
<% if File.exist?('/bin/bash') %>
/some/bin/say_bash_exists
<% else %>
<% rubyerror "bash is missing" %. ## hypothesizing, but if bash dne, you cannot 'echo', or assume any other shell that can, no?
<% end %>
```

**Comment by yeoldegrove** [comment]
A more real live scnario would the rendering of an ERB file in the chef config management case:

e.g.
```
<% if { node['virtualization']['systems']['docker'] == 'guest' } %>
echo "I am running in docker"
<% else %>
echo "I am NOT running in docker"
<% end %>
```

Could one implement a switch to turn on/off validating of such template systems?
